### PR TITLE
Add docs clarifying list-doc and install-doc

### PR DESCRIPTION
**Pull request by rectalogic**
Fixes https://github.com/miurahr/aqtinstall/issues/753

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -317,6 +317,44 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+archives for a given version by default:
+
+.. code-block:: console
+
+    $ aqt list-tool mac 6.6.1
+    qdoc qmake qt5 qtassistant qtcmake qtconcurrent qtcore qtdbus qtdesigner qtdistancefieldgenerator qtdoc qtgui qthelp qtlabsplatform qtlinguist qtnetwork qtopengl qtplatformintegration qtprintsupport qtqml qtqmlcore qtqmlmodels qtqmltest qtqmlworkerscript qtqmlxmllistmodel qtquick qtquickcontrols qtquickdialogs qtsql qtsvg qttestlib qtuitools qtwaylandcompositor qtwidgets qtxml
+
+This is what will be installed by default, you can select a subset of archives
+with ``--archives``.
+``--modules`` will list additional documentation modules that can be installed:
```
Comment (ddalcino) [inline docs/getting_started.rst:333]:
```suggestion
This is what will be installed by default when you run ``aqt install-doc mac 6.6.1`` 
without any other flags or arguments. 
You can select a subset of archives with the ``--archives`` flag.
The ``--modules`` flag will list additional documentation modules that can be installed:
```
This is more of a writing-style / grammar suggestion. IMHO it's worthwhile to be as clear as possible here. I'm certainly not the best writer around, but I think we can try to make this a little better.

Mozi is pretty good at this stuff. @pzhlkj6612, any suggestions?

File: docs/getting_started.rst [hunk 2]
```diff
@@ -317,6 +317,44 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1 --modules
+    qt3d qt5compat qtactiveqt qtbluetooth qtcharts qtdatavis3d qtgraphs qtgrpc qthttpserver qtimageformats qtlocation qtlottie qtmultimedia qtnetworkauth qtnfc qtpdf qtpositioning qtquick3d qtquick3dphysics qtquickeffectmaker qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview
+
+``--archives`` and ``--modules`` can be used together.
+For example to only install the docs for ``qtquick`` and ``qt3d``,
```
Comment (ddalcino) [inline docs/getting_started.rst:342]:
```suggestion
For example, to only install the docs for ``qtquick`` and ``qt3d``,
```

File: docs/getting_started.rst [hunk 3]
```diff
@@ -317,6 +317,44 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+with ``--archives``.
+``--modules`` will list additional documentation modules that can be installed:
+
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1 --modules
+    qt3d qt5compat qtactiveqt qtbluetooth qtcharts qtdatavis3d qtgraphs qtgrpc qthttpserver qtimageformats qtlocation qtlottie qtmultimedia qtnetworkauth qtnfc qtpdf qtpositioning qtquick3d qtquick3dphysics qtquickeffectmaker qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview
+
+``--archives`` and ``--modules`` can be used together.
```
Comment (ddalcino) [inline docs/getting_started.rst:341]:
```suggestion
``The --archives`` and ``--modules`` flags can be used together.
```

**Review by ddalcino — COMMENTED** [review]
Thank you for the PR! This is definitely an improvement.

Please see the suggested changes below. These are my initial thoughts on how to make this section a little more clear; they are not perfect, but I'll bet we can collectively come up with something better.

**Comment by rectalogic** [comment]
@ddalcino thanks. So one thing I noticed is `list-doc` does not accept `--archives`. So for example the `qtmultimedia` module consists of 2 archives - `qtmultimedia` and `qtspatialaudio`.

So if I only want the `qtmultimedia` docs, I need to do `install-doc mac 6.6.1 --modules qtmultimedia --archives qtmultimedia` - if I just do `install-doc mac 6.6.1 --modules qtmultimedia` then it installs all the default archives plus `qtmultimedia` and `qtspatialaudio` docs.

But I don't see a way to figure this out from `list-doc`. I would just have to install `--modules qtmultimedia` and realize it is installing multiple modules, then run it again filtering down the archives I guess.

So basically the archives that make up a module are hidden from `list-doc` - it can only show the default set of archives, or the list of modules. But not the archives that a module consists of.

**Comment by ddalcino** [comment]
You are correct. In general, there is only a single archive for each module, except for the base module and the `debug_info` module, which only exists for some platforms when installing Qt (not src/doc/examples).

There are a few edge cases, like this one, where a module contains 2 or more archives.

The purpose of the `--archives` command is really to help reduce download times and data usage, and I think it does that well enough. I don't think it's worth the maintenance cost to enable it for a couple edge cases here to save a couple kilobytes.

You're welcome to document this as well if it really bothers you.

File: docs/getting_started.rst [hunk 4]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+You can select a subset of archives with the ``--archives`` flag.
+The ``--modules`` flag will list additional documentation modules that can be installed:
+
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1 --modules
+    qt3d qt5compat qtactiveqt qtbluetooth qtcharts qtdatavis3d qtgraphs qtgrpc qthttpserver qtimageformats qtlocation qtlottie qtmultimedia qtnetworkauth qtnfc qtpdf qtpositioning qtquick3d qtquick3dphysics qtquickeffectmaker qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview
+
+``The --archives`` and ``--modules`` flags can be used together.
```
Comment (ddalcino) [inline docs/getting_started.rst:342]:
```suggestion
The ``--archives`` and ``--modules`` flags can be used together.
```

Oops! Sorry, that was my fault.

File: docs/getting_started.rst [hunk 5]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
     $ aqt install-qt linux desktop 6.2.0 wasm_32 -m qtcharts qtnetworkauth --autodesktop
 
 
+Installing Docs
+---------------
+
+The :ref:`aqt list-doc <list-doc command>` command lists documentation
+archives for a given version by default:
```
Comment (pzhlkj6612) [inline docs/getting_started.rst:324]: I think we could not emphasize the "default".

```suggestion
The :ref:`aqt list-doc <list-doc command>` command lists documentation
archives for a given Qt version:
```

File: docs/getting_started.rst [hunk 6]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1
+    qdoc qmake qt5 qtassistant qtcmake qtconcurrent qtcore qtdbus qtdesigner qtdistancefieldgenerator qtdoc qtgui qthelp qtlabsplatform qtlinguist qtnetwork qtopengl qtplatformintegration qtprintsupport qtqml qtqmlcore qtqmlmodels qtqmltest qtqmlworkerscript qtqmlxmllistmodel qtquick qtquickcontrols qtquickdialogs qtsql qtsvg qttestlib qtuitools qtwaylandcompositor qtwidgets qtxml
+
+This is what will be installed by default when you run ``aqt install-doc mac 6.6.1`` 
+without any other flags or arguments. 
+You can select a subset of archives with the ``--archives`` flag.
```
Comment (pzhlkj6612) [inline docs/getting_started.rst:333]: I think there is no need to emphasize the "default".

```suggestion
All of the above archives will be installed when you run ``aqt install-doc mac 6.6.1`` 
without any other flags or arguments. 
You can select a subset of them with the ``--archives`` flag.
```

File: docs/getting_started.rst [hunk 7]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+The ``--modules`` flag will list additional documentation modules that can be installed:
```
Comment (pzhlkj6612) [inline docs/getting_started.rst:334]: New line; tense.

```suggestion

The ``--modules`` flag lists additional documentation modules that can be installed:
```

File: docs/getting_started.rst [hunk 8]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1 --modules
+    qt3d qt5compat qtactiveqt qtbluetooth qtcharts qtdatavis3d qtgraphs qtgrpc qthttpserver qtimageformats qtlocation qtlottie qtmultimedia qtnetworkauth qtnfc qtpdf qtpositioning qtquick3d qtquick3dphysics qtquickeffectmaker qtquicktimeline qtremoteobjects qtscxml qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel qtwebengine qtwebsockets qtwebview
```
Comment (pzhlkj6612) [inline docs/getting_started.rst:340]: Wrap the long line.

```suggestion
    $ aqt list-doc mac 6.6.1 --modules
    qt3d qt5compat qtactiveqt qtbluetooth qtcharts qtdatavis3d qtgraphs qtgrpc qthttpserver
    qtimageformats qtlocation qtlottie qtmultimedia qtnetworkauth qtnfc qtpdf qtpositioning
    qtquick3d qtquick3dphysics qtquickeffectmaker qtquicktimeline qtremoteobjects qtscxml
    qtsensors qtserialbus qtserialport qtshadertools qtspeech qtvirtualkeyboard qtwebchannel
    qtwebengine qtwebsockets qtwebview
```

File: docs/getting_started.rst [hunk 9]
```diff
@@ -317,6 +317,45 @@ You can do this automatically with the ``--autodesktop`` flag:
...
+Installing Docs
+---------------
+
+The :ref:`aqt list-doc <list-doc command>` command lists documentation
+archives for a given version by default:
+
+.. code-block:: console
+
+    $ aqt list-doc mac 6.6.1
+    qdoc qmake qt5 qtassistant qtcmake qtconcurrent qtcore qtdbus qtdesigner qtdistancefieldgenerator qtdoc qtgui qthelp qtlabsplatform qtlinguist qtnetwork qtopengl qtplatformintegration qtprintsupport qtqml qtqmlcore qtqmlmodels qtqmltest qtqmlworkerscript qtqmlxmllistmodel qtquick qtquickcontrols qtquickdialogs qtsql qtsvg qttestlib qtuitools qtwaylandcompositor qtwidgets qtxml
```
Comment (pzhlkj6612) [inline docs/getting_started.rst:329]: Wrap the long line.

```suggestion
    $ aqt list-doc mac 6.6.1
    qdoc qmake qt5 qtassistant qtcmake qtconcurrent qtcore qtdbus qtdesigner
    qtdistancefieldgenerator qtdoc qtgui qthelp qtlabsplatform qtlinguist qtnetwork
    qtopengl qtplatformintegration qtprintsupport qtqml qtqmlcore qtqmlmodels qtqmltest
    qtqmlworkerscript qtqmlxmllistmodel qtquick qtquickcontrols qtquickdialogs qtsql
    qtsvg qttestlib qtuitools qtwaylandcompositor qtwidgets qtxml
```

**Review by pzhlkj6612 — COMMENTED** [review]
Wording wording.

**Review by miurahr — COMMENTED** [review]
Confirmed suggested wording improvements.

**Comment by pzhlkj6612** [comment]
A GitHub pro tip for @rectalogic:

We are able to apply a batch of changes during code reviewing: [Incorporating feedback in your pull request - GitHub Docs](https://docs.github.com/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request).

**Review by pzhlkj6612 — APPROVED** [review]
I didn't test the code, but have read the docs. It looks OK to me.

**Review by ddalcino — APPROVED** [review]
LGTM. I ran the code and it does exactly what the code-block sections say they will.
